user/syscall_test.c

```diff
@@ -0,0 +1,25 @@
+#include <stddef.h>
+
+int write(int fd, const char *buf, size_t count) {
+  int retval, errval;
+  asm volatile("move $a0, %2\n"
+               "move $a1, %3\n"
+               "move $a2, %4\n"
+               "li $v0, 5\n" /* SYS_WRITE */
+               "syscall\n"
+               "move %0, $v0\n"
+               "move %1, $v1\n"
+               : "=r"(retval), "=r"(errval)
+               : "r"(fd), "r"(buf), "r"(count)
+               : "4", "5", "6", "2", "3");
```

Comment:
Ok. My bad. I got confused by the fact you use register numbers and not symbolic names.

To avoid readability issues I suggest to rewrite the routine in pure assembly.